drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+) -> capa.rules.RuleSet:
+    # this method filters the given rules folder following arbitrary logic (specified by filter_function)
+    rules = list(rules.rules.values())
+    reduced_rules = list(filter(filter_function, rules))
+
+    # filter the given rules folder using the specified boolean function
+    filtered_rules = set()
+    for rule in reduced_rules:
+        filtered_rules.update(
+            set(capa.rules.get_rules_and_dependencies(rules, rule.name))
+        )
+
+    # generate the new ruleset, and return it
+    return capa.rules.RuleSet(list(filtered_rules))
+
+
+def get_malware_processes(
+    metadata_path: Path, inject_path: Path, pstree_path: Path
+) -> Optional[List[str]]:
+    # this method gets all the pids in the Drakvuf report that are associated with malware
+    with metadata_path.open("r") as f:
+        # we use the metadata file to get the analysis' start command
+        metadata = orjson.loads(f.read())
+
+    with inject_path.open("r") as f:
+        # we use the injected processes log to get the malware process' parent pid
+        injected_processes = [orjson.loads(line) for line in f]
+
+    with pstree_path.open("r") as f:
+        # we use the process tree to get all of the malware process' child processes
+        pstree = orjson.loads(f.read())
+
+    # make sure we have the right inject.log entry (using the metadata.json file)
+    malware_injection_log: str = next(
+        filter(
+            lambda line: line["ProcessName"] == metadata["start_command"],
+            injected_processes,
+        )
+    )
+
+    # get the parent malware process' pid and process name
+    malware_pid: int = malware_injection_log["InjectedPid"]
+    malware_procname: str = malware_injection_log["ProcessName"]
+
+    # find the malware process' sub-tree in the overall process tree
+    try:
+        malware_process: Dict = find_process_in_pstree(
+            pstree, malware_pid, malware_procname
+        )
+    except ValueError:
+        # malware pid not found, analyze entire log instead
+        return None
+
+    # get all child processes of the malware process
+    malware_pids = get_all_child_processes(malware_process)
+
+    # return a list of all spawned malware processes (does not include process injection)
+    return malware_pids
+
+
+def get_drakvuf_feature_extractor(
+    calls: Iterator[Dict],
+) -> capa.features.extractors.drakvuf.extractor.DrakvufExtractor:
+    # wrapper routing for initializing the Drakvuf feature extractor
+    return capa.features.extractors.drakvuf.extractor.DrakvufExtractor.from_report(
+        calls
+    )
+
+
+def decode_json_lines(fd: BinaryIO) -> Iterator[Dict]:
+    # read and decode json lines into a list of dictionaries
+    for line in fd:
+        try:
+            # we use orjson for a small performance improvement
+            line_s = line.strip()
+            obj = orjson.loads(line_s)
+            yield obj
```

Comment:
no need to be so verbose
```suggestion 
            yield orjson.loads(line.strip())
```